### DISCUSSION TITLE
Discrete inner-product on non-uniform mesh

### DISCUSSION THREAD

**Discussion by MarcoArtiano** [xcompact3d/Incompact3d#170] [Q&A] [answered]
Hi guys, I'm trying to write the same post-process that appears in BC-TGV.f90, but for cases where the mesh is non uniform, for example, x and z are uniform and y is stretched. 

```
       do k=1,xsize3
          do j=1,xsize2
             do i=1,xsize1
                temp1=temp1+zpfive*((ux1(i,j,k))**2+(uy1(i,j,k))**2+(uz1(i,j,k))**2)
             enddo
          enddo
       enddo
       call MPI_ALLREDUCE(temp1,eek,1,real_type,MPI_SUM,MPI_COMM_WORLD,code)
       eek=eek/(nxc*nyc*nzc)
```

I thought to modify that as follows
   
   ```
  temp1=zero
       do k=1,xsize3
          do j=1,xsize2
            if (j.eq.1) then
                  dyv = (yp(2)-yp(1))*half
            elseif (j.eq.xsize2) then
               dyv = (yp(xsize2)-yp(xsize2-1))*half
            else
               dyv = (yp(j)+yp(j-1))*half
            endif
              do i=1,xsize1
                 temp1=temp1+dyv*zpfive*((ux1(i,j,k))**2+(uy1(i,j,k))**2+(uz1(i,j,k))**2)
             enddo
          enddo
       enddo
       call MPI_ALLREDUCE(temp1,Ec,1,real_type,MPI_SUM,MPI_COMM_WORLD,code)
     Ec = Ec/(nxc*nzc)
```
But dunno if this is correct or not, or should I go for like computing the mean value in the cell center and multiplying for the volume of the cell, so I don't have to take half volume for the points collocated at the edge of the domain. 
 
Thanks so much in advance

**Comment by mathrack** [comment, marked as answer]
Hello,

You can use something based on the estimation of the averaged velocity available in the channel flow subroutines : https://github.com/xcompact3d/Incompact3d/blob/master/src/BC-Channel-flow.f90#L200

Best regards,
Mathrack

**Comment by MarcoArtiano** [comment]
Oh thank you so much, I missed that !
